views-core/src/main/java/io/micronaut/views/ViewsFilter.java

```diff
@@ -113,14 +107,9 @@ public int getOrder() {
                         }
 
                         ViewsRenderer viewsRenderer = optionalViewsRenderer.get();
-                        // We treat Map<String, Object> the same as a model view as long as
-                        // the method includes a view annotation:
-                        if (body instanceof ModelAndView || body instanceof Map) {
-                            Map<String, Object> context = populateModel(request, viewsRenderer, body);
-                            modelAndView  = processModelAndView(request, view, context);
-                        } else {
-                            modelAndView = new ModelAndView<>(view, body);
-                        }
+                        ModelAndView<?> modelAndView  = new ModelAndView<>(view, (body instanceof ModelAndView || body instanceof Map) ?
+                                    populateModel(request, viewsRenderer, body) : body);
```

Comment:
This one is tricky to read, it would be better if you extract it and use a variable

Reply:
I've reverted to the old if/else way of writing this code.